src/commands/githubActions/bumpUpstream/index.ts

```diff
@@ -55,180 +49,178 @@ export const gaBumpUpstream: CommandModule<
+      const latestRelease = releases[0];
+      if (!latestRelease) throw Error(`No release found for ${upstreamRepo}`);
 
-    const argName = upstreamArgs[i];
-    const newVersion = latestRelease.tag_name;
+      const newVersion = latestRelease.tag_name;
+      if (!isValidRelease(newVersion)) {
+        console.log(`This is not a valid release (probably a release candidate) - ${upstreamRepo}: ${newVersion}`);
+        continue;
+      }
 
-    if (isUndesiredRealease(newVersion)) {
-      console.log(`This is a realease candidate - ${repoSlug}: ${newVersion}`);
-      return;
+      upstreamRepoVersions[upstreamArg] = { repo, repoSlug: upstreamRepo, newVersion };
+      console.log(`Fetch latest version(s) - ${upstreamRepo}: ${newVersion}`);
     }
-    upstreamRepoVersions.set(argName, { repo, repoSlug, newVersion });
-
-    console.log(`Fetch latest version(s) - ${repoSlug}: ${newVersion}`);
+  } catch (e) {
+    console.error("Error fetching upstream repo versions:", e);
+    throw e;
   }
 
-  // Compute branch name
-  const branch =
-    branchNameRoot +
-    Array.from(upstreamRepoVersions.values())
-      .map(({ repo, newVersion }) => `${repo}@${newVersion}`)
-      .join(",");
-  const branchRef = `refs/heads/${branch}`;
-
-  // Get current upstream version
-  const thisRepo = Github.fromLocal(dir);
-  const repoData = await thisRepo.getRepo();
-  const remoteBranchExists = await thisRepo.branchExists(branch);
-  const localBranchExists = await getLocalBranchExists(branch);
-  const branchExists = remoteBranchExists || localBranchExists;
-
-  if (branchExists) {
-    console.log(`Branch ${branch} already exists`);
-    return;
-    // TODO, do some checking
-  }
+  return upstreamRepoVersions;
+}
 
-  // index by repoSlug, must be unique
-  const versionsToUpdateMap = new Map<string, VersionToUpdate>();
-
-  for (const [serviceName, service] of Object.entries(compose.services))
-    if (typeof service.build === "object" && service.build.args)
-      for (const [argName, argValue] of Object.entries(service.build.args)) {
-        const upstreamRepoVersion = upstreamRepoVersions.get(argName);
-        if (!upstreamRepoVersion) continue;
-
-        const currentVersion = argValue;
-        const { repoSlug, newVersion } = upstreamRepoVersion;
-        if (currentVersion === newVersion) continue;
-
-        // Update current version
-        compose.services[serviceName].build = {
-          ...service.build,
-          args: {
-            ...service.build.args,
-            [argName]: newVersion
-          }
-        };
-        // Use a Map since there may be multiple matches for the same argName
-        versionsToUpdateMap.set(repoSlug, {
-          repoSlug,
-          newVersion,
-          currentVersion
-        });
-      }
+function updateComposeVersions(dir: string, compose: Compose, upstreamRepoVersions: UpstreamRepoMap): ComposeVersionsToUpdate {
+  const newCompose = JSON.parse(JSON.stringify(compose)); // Deep copy
```

Comment:
why is this needed?

Reply:
It is used to update the manifest and to open the PR